git-extra/aliases.sh

```diff
@@ -10,7 +10,7 @@ xterm*)
 	# The following programs are known to require a Win32 Console
 	# for interactive usage, therefore let's launch them through winpty
 	# when run inside `mintty`.
-	for name in node ipython php php5 psql python2.7
+	for name in node ipython php php5 psql python2.7 docker vault ${GITBASH_WINPTY}
```

Comment:
```suggestion
	for name in node ipython php php5 psql python2.7 docker vault
```

Reply:
I added the GITBASH_WINPTY variable, as it would have been invaluable when I was building Boot2Docker and the installers that became Docker Toolbox - I would have been able to set the installed shell environment to add winpty to all calls :)

I'm happy to remove it, I just figured it'd be good to talk about it first

Reply:
> I added the GITBASH_WINPTY variable, as it would have been invaluable when I was building Boot2Docker and the installers that became Docker Toolbox

I see, in Docker it is easy to set environment variables, and not that easy to inject a `~/.bash_profile`...

Still, the variable needs a better name. The name needs to convey that those commands will be aliased to use `winpty` in order to get a real Win32 Console.

Reply:
how about `GITBASH_ADD_WINPTY_TO_COMMANDS` or `GITBASH_COMMANDS_TO_ADD_WINPTY_TO` ?

yup names are hard

Reply:
How about `NEED_WIN32_CONSOLE`?

Reply:
(I am trying to get away from implementation details in names, and rather giving helpful hints what it can be used for.)

Reply:
Please consider all invocations of `docker` are not meant to be interactive, even from the command line.
Piping in and out of a Docker container does not work if wrapped with winpty.
See rprichard/winpty#101 among a few others.

Reply:
@SvenDowideit the current version still gets hung up on an implementation detail, and @JulienCochuyt's comment about `docker` seems quite relevant to me... Care to consider?

Reply:
yeah, I think this is a mega-hard issue with no great answers.

I'm basically avoiding it for now by switching off Windows :( so I don't think I can be much help this year (I switch around regularly to have more of an idea of user problems)

Reply:
@SvenDowideit so should we close this PR?